src/components/AuthValidation/AuthValidation.js

```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { useNavigate } from "react-router";
+import { useLogedUser } from "../../context/UserContext";
+
+const AuthValidation = ({ children }) => {
+  let { user, loading } = useLogedUser();
+  const navigate = useNavigate();
+
+  const isAuth = user?.token;
+
+  if (loading) {
+    return <h1>Loading..</h1>;
+  }
+  return isAuth ? <>{children}</> : navigate("/");

```

Comment:
Hacer un return de `navigate("/")` lo veo riesgoso, ya que si en algún momento esa función regresa algo que no esperábamos, podría provocar que la aplicación se rompa en ese momento, tratando de renderizar ese resultado.
```suggestion
else{
   if (!isAuth) navigate("/");
}
  return isAuth ? <>{children}</> : null;

```

Reply:
listo ya se realizaron los cambios